tests/streamlit_ui_tests/streamlit.spec.ts

```diff
@@ -320,6 +335,161 @@ test.describe('Home Tab Buttons', () => {
+    // We test merge functionality elsewhere, so we skip here
+  });
+
+  test('Test Concat', async ({ page }) => {
+    const mito = await getMitoFrameWithTestCSV(page);
+    await importCSV(page, mito, 'test.csv');
+    await clickTab(page, mito, 'Insert');
+
+    await clickButtonAndAwaitResponse(page, mito, 'Concat');
+
+    await expect(mito.getByText('Concatenate Sheet')).toBeVisible();
+    // We test concat functionality elsewhere, so we skip here
+
+  });
+
+  test('Test Anti-merge', async ({ page }) => {
+    const mito = await getMitoFrameWithTestCSV(page);
+    await importCSV(page, mito, 'test.csv');
+    await clickTab(page, mito, 'Insert');
+
+    await clickButtonAndAwaitResponse(page, mito, 'Anti-merge');
+
+    await expect(mito.getByText('Merge Dataframes')).toBeVisible();
+    await expect(mito.getByText('unique in left')).toBeVisible();
+    // We test anti-merge functionality elsewhere, so we skip here
+  });
+
+  test('Test Graph', async ({ page }) => {
+    const mito = await getMitoFrameWithTestCSV(page);
+    await clickTab(page, mito, 'Insert');
+
+    await clickButtonAndAwaitResponse(page, mito, { name: 'Graph', exact: true });
+
+    await expect(mito.getByText('Setup Graph')).toBeVisible();
+    // We test graph functionality elsewhere, so we skip here
+  });
+
+  test('Test Graph (scatter)', async ({ page }) => {
+    const mito = await getMitoFrameWithTestCSV(page);
+    await clickTab(page, mito, 'Insert');
+
+    await clickButtonAndAwaitResponse(page, mito, { name: 'Create an interactive scatter plot.' });
+
+    await expect(mito.getByText('Setup Graph')).toBeVisible();
+    await expect(mito.getByText('Scatter')).toBeVisible();
+  });
+
+  test('Test Graph (line)', async ({ page }) => {
+    const mito = await getMitoFrameWithTestCSV(page);
+    await clickTab(page, mito, 'Insert');
+
+    await clickButtonAndAwaitResponse(page, mito, { name: 'Create an interactive line graph.' });
+
+    await expect(mito.getByText('Setup Graph')).toBeVisible();
+    await expect(mito.getByText('line', {exact: true})).toBeVisible();
+
+  });
+
+})
+
+
+test.describe('Code Tab Buttons', () => {
+
+  test('Test Copy Code', async ({ page }) => {
+    const mito = await getMitoFrameWithTestCSV(page);
+    await clickTab(page, mito, 'Code');
+
+    await mito.getByRole('button', { name: 'Copy Code' }).click();
+    // Check the code is copied to the clipboard
+    // TODO: There are some bugs with Playwrite, that make it hard to check the clipboard contents
+  });
+
+  test('Test Configure Code', async ({ page }) => {
+    const mito = await getMitoFrameWithTestCSV(page);
+    await clickTab(page, mito, 'Code');
+
+    await mito.getByRole('button', { name: 'Configure Code' }).click();
+    await checkOpenTaskpane(mito, 'Generated Code Options');
+
+    await mito.locator('.toggle').first().click();
+    await mito.getByRole('textbox').fill('new name');
+    // Wanna check some output
```

Comment:
Doesn't look like you're checking the output?

Reply:
Yeah, per comment I think we can ignore the code tests.